trl/trainer/grpo_trainer.py

```diff
@@ -845,23 +856,80 @@ def _generate_and_score_completions(
                     reward_inputs = super()._prepare_inputs(reward_inputs)
                     with torch.inference_mode():
                         rewards_per_func[:, i] = reward_func(**reward_inputs).logits[:, 0]  # Shape (B*G,)
+                        has_valid_rewards = has_valid_rewards | ~torch.isnan(rewards_per_func[:, i])
                 else:
                     # Repeat all input columns (but "prompt" and "completion") to match the number of generations
                     keys = [key for key in inputs[0] if key not in ["prompt", "completion"]]
                     reward_kwargs = {key: [example[key] for example in inputs] for key in keys}
                     output_reward_func = reward_func(prompts=prompts, completions=completions, **reward_kwargs)
-                    rewards_per_func[:, i] = torch.tensor(output_reward_func, dtype=torch.float32, device=device)
+                    
+                    # Handle None values in the reward function output
+                    if output_reward_func is None:
+                        # If the entire function returns None, skip this reward function
+                        continue
```

Comment:
the function is expected to return a list (even containing only Nones), so we can remove it
```suggestion
```